identity-apps-core/apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -129,57 +175,168 @@
+            RecoveryApiV2 recoveryApiV2 = new RecoveryApiV2();
+            String recoveryCode = request.getParameter("recoveryCode") != null 
+                                ? request.getParameter("recoveryCode") 
+                                : IdentityManagementEndpointUtil.getStringValue(request.getAttribute("recoveryCode"));
+
+            String usernameRecoveryOption = request.getParameter("usernameRecoveryOption") != null 
+                        ? request.getParameter("usernameRecoveryOption") 
+                        : IdentityManagementEndpointUtil.getStringValue(request.getAttribute("usernameRecoveryOption"));
+
+            String isUserFoundParam = request.getParameter("isUserFound");
+            Boolean isUserFound = (isUserFoundParam != null) 
+                                ? Boolean.parseBoolean(isUserFoundParam) 
+                                : IdentityManagementEndpointUtil.getBooleanValue(request.getAttribute("isUserFound"));
+
+            String recoveryChannelType = null;
+            String recoveryChannelId = null;
+            
+            // Extract the recovery channel name and id from the recovery option.
+            if(usernameRecoveryOption != null){
+                String[] parts = usernameRecoveryOption.split(":");
+                recoveryChannelId = parts[0];
+                recoveryChannelType = parts[1];
+            }
+            
+            request.setAttribute("recoveryChannelType", recoveryChannelType);
+ 
+            // Sending the notification if we only found a user.
+            if (isUserFound){
+                RecoveryRequest recoveryRequest = new RecoveryRequest();
+                recoveryRequest.setRecoveryCode(recoveryCode);
+                recoveryRequest.setChannelId(recoveryChannelId);
+
+                Map<String, String> requestHeaders = new HashedMap();
+                if (request.getParameter("g-recaptcha-response") != null) {
+                    requestHeaders.put("g-recaptcha-response", request.getParameter("g-recaptcha-response"));
+                }
 
-            IdentityManagementEndpointUtil.addErrorInformation(request, e);
-            request.getRequestDispatcher("recoveraccountrouter.do").forward(request, response);
+                try{
+                    recoveryApiV2.recoverUsername(recoveryRequest, tenantDomain, requestHeaders);
+                    
+                } catch (ApiException e) {
+                    IdentityManagementEndpointUtil.addErrorInformation(request, e);
+                    request.getRequestDispatcher("recoveraccountrouter.do").forward(request, response);
+                    return;
+                }
+            }
+            
+            request.getRequestDispatcher("username-recovery-complete.jsp").forward(request, response);
             return;
+
+        } else {
+            request.setAttribute("errorMsg", "Invalid username recovery stage. recovery Stage: "+recoveryStage);
```

Comment:
shouldn't we send the errorMsg as a i18n string?

Reply:
Fixed by https://github.com/wso2/identity-apps/pull/7005/commits/7e9237b00aa81448f1ba9d5a8afb4695d09a3fae